hw/drivers/sensors/dps368/src/dps368.c

```diff
@@ -0,0 +1,1026 @@
+         coeffs->C10 = coeffs->C10 - POW_2_20;
+
+     coeffs->C01 = (read_buffer[9] + (read_buffer[8]<<8));
+     if(coeffs->C01 > ((POW_2_15_MINUS_1)))
+         coeffs->C01 = coeffs->C01 - POW_2_16;
+
+     coeffs->C11 = (read_buffer[11] + (read_buffer[10]<<8));
+     if(coeffs->C11 > ((POW_2_15_MINUS_1)))
+         coeffs->C11 = coeffs->C11 - POW_2_16;
+
+     coeffs->C20 = (read_buffer[13] + (read_buffer[12]<<8));
+     if(coeffs->C20 > ((POW_2_15_MINUS_1)))
+         coeffs->C20 = coeffs->C20 - POW_2_16;
+
+     coeffs->C21 = (read_buffer[15] + (read_buffer[14]<<8));
+     if(coeffs->C21 > ((POW_2_15_MINUS_1)))
+         coeffs->C21 = coeffs->C21 - POW_2_16;
+
+     coeffs->C30 = (read_buffer[17] + (read_buffer[16]<<8));
+     if(coeffs->C30 > ((POW_2_15_MINUS_1)))
+         coeffs->C30 = coeffs->C30 - POW_2_16;
+
+     memset(read_buffer, 0, sizeof(IFX_DPS368_COEF_LEN));
+
+     rc = dps368_read_regs(itf, IFX_DPS368_TMP_COEF_SRCE_REG_ADDR, read_buffer,
+             IFX_DPS368_TMP_CFG_REG_LEN);
+
+     if (rc) {
+         return rc;
+     }
+
+    if ((read_buffer[0] >> IFX_DPS368_TMP_COEF_SRCE_REG_POS) & 1) {
+        *src_t = TMP_EXT_MEMS;
+    } else {
+        *src_t = TMP_EXT_ASIC;
+    }
+
+     return 0;
+}
+
+/**
+ * Choose appropriate scalling factor for given OSR
+ *
+ * @param selected osr
+ * @return chosen scaling coefficient for given osr
+ */
+static dps3xx_scaling_coeffs_e
+dps368_get_scaling_coef (dps3xx_osr_e osr)
+{
+    dps3xx_scaling_coeffs_e scaling_coeff;
+
+    switch (osr){
+
+    case OSR_1:
+        scaling_coeff = OSR_SF_1;
+        break;
+    case OSR_2:
+        scaling_coeff = OSR_SF_2;
+        break;
+    case OSR_4:
+        scaling_coeff = OSR_SF_4;
+        break;
+    case OSR_8:
+        scaling_coeff = OSR_SF_8;
+        break;
+    case OSR_16:
+        scaling_coeff = OSR_SF_16;
+        break;
+    case OSR_32:
+        scaling_coeff = OSR_SF_32;
+        break;
+    case OSR_64:
+        scaling_coeff = OSR_SF_64;
+        break;
+    case OSR_128:
+        scaling_coeff = OSR_SF_128;
+        break;
+    default:
+        scaling_coeff = OSR_SF_1;
+        break;
+    }
+
+    return scaling_coeff;
+}
+
+/**
+ * post init sequence
+ *
+ * @param ptr sensor interface
+ * @return 0 on success else non 0 val
+ */
+static int
+dps368_set_oem_parameters(struct sensor_itf *itf)
+{
+    int rc;
+
+    if ((rc = dps368_write_reg(itf, (uint8_t) 0x0E, (uint8_t) 0xA5))) {
+        return rc;
+    }
+
+    if ((rc = dps368_write_reg(itf, (uint8_t) 0x0F, (uint8_t) 0x96))) {
+        return rc;
+    }
+
+    if ((rc = dps368_write_reg(itf, (uint8_t) 0x62, (uint8_t) 0x02))) {
+        return rc;
+    }
+
+    if ((rc = dps368_write_reg(itf, (uint8_t) 0x0E, (uint8_t) 0x00))) {
+        return rc;
+    }
+
+    if ((rc = dps368_write_reg(itf, (uint8_t) 0x0F, (uint8_t) 0x00))) {
+        return rc;
+    }
+
+    DPS368_LOG(INFO,"DPS368:OEM Parameters are set\n");
+
+    return 0;
+
+}

```

Comment:
@kasjer The code snippet you are referring to actually unlocks the sensor by writing a signature 0xa5 and 0x96 on respective addresses. Then it sets register to reflect a more accurate/tweaked configuration that is to be used in order to get more accurate reading w.r.t. temperature etc. Finally it locks back the register space which is otherwise protected from end user.
The same driver can be used for DPS310/368/280 etc, and for those variants this code snippet would be helpful. Of course as its internal tweak, its not documented officially. 
So its is up to you to decide whether you would like to keep it or not.
Maybe a good idea is to create another driver patch for DPS310/280 where it is more appropriate to keep this.

Reply:
@saumitra-chafekar thanks for explanation, I will make this part optional so it could be included by user if needed.
Do you know of other parts that could share driver code (by Infineon or someone else) that may not need this tweak?